gamemodes/terrortown/gamemode/client/cl_scoring.lua

```diff
@@ -500,15 +499,77 @@ function CLSCORE:BuildHilitePanel(dpanel)
 	end
 end
 
+---
+-- Converts the wintype into a title object to be used with @{CLSCORE:BuildHilitePanel}.
+--
+-- @internal
+-- @realm client
+-- @param string|number wintype
+-- @return table The title object table
+function CLSCORE:BuildTitle(wintype)
+	-- wintype can be both a WIN_ var or the team name directly, so we need to convert it to a teamname
+	local winnerTeam = TEAM_INNOCENT
+	if wintype == WIN_TIMELIMIT or wintype == WIN_INNOCENT then
+		winnerTeam = TEAM_INNOCENT
+	elseif wintype == WIN_TRAITOR then
+		winnerTeam = TEAM_TRAITOR
+	elseif wintype == WIN_NONE then
+		-- Special case, that should not be reachable, as WIN_NONE usually prevents a round from ending
+		-- If it should, we will show a tie.
+		return {
+			Text = "hilite_win_bees",
+			BoxColor = TEAMS[TEAM_INNOCENT].color,
+			TextColor = COLOR_WHITE,
+			BackgroundColor = Color(50, 50, 50, 255)
+		}
+	elseif isstring(wintype) then
+		winnerTeam = wintype
+	end
+
+	return {
+		Text = "hilite_win_" .. winnerTeam,
+		BoxColor = TEAMS[winnerTeam].color,
+		TextColor = COLOR_WHITE,
+		BackgroundColor = Color(50, 50, 50, 255)
+	}
+end
+
 ---
 -- Displays the score @{Panel} for the local @{Player}
 -- @realm client
 function CLSCORE:ShowPanel()
+	if IsValid(self.Panel) then
+		self:ClearPanel()
+	end
+
+	local wintype
 	local margin = 15
 	local dpanel = vgui.Create("DFrame")
-	local w, h = 700, 500
+	local starttime = self.StartTime
+	local endtime = starttime
+	local events = self.Events
 
-	dpanel:SetSize(700, 500)
+	for i = #events, 1, -1 do
+		local e = events[i]
+		if e.id == EVENT_FINISH then
+			endtime = e.t
+			wintype = e.win
+			break
+		end
+	end
+
+	local title = self:BuildTitle(wintype)
+
+	-- size the panel based on the win text w/ 88px horizontal padding and 44px veritcal padding
+	surface.SetFont("WinHuge")
+	local w, h = surface.GetTextSize(T(title.Text))
+
+	MsgN("Calculated width of text = " .. w)
+
+	-- w + padding (100) + DPropertySheet padding (8) + winlbl padding (30) + offset margin (margin * 2) + size margin (margin)
+	w, h = math.max(700, w + 138 + margin * 3), 500
```

Comment:
`700` is the max size. Just two ideas, we may discuss, even for the later rework: Shouldn't this be a dynamic value to support responsive design (`500` in height as well)? And shouldn't there be a minimum value as well?

Reply:
math.max will just choose the highest number and does not limit the width to 700, instead it could rather be understood as a minimum.